sequence_check_digit/tests/test_check_digit.py

```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Creu Blanca <https://creublanca.es/>
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+
+from odoo.tests import common
+import logging
+from odoo.exceptions import ValidationError
+try:
+    from stdnum.iso7064 import mod_97_10
+    from stdnum.iso7064 import mod_37_2, mod_37_36
+    from stdnum.iso7064 import mod_11_2, mod_11_10
+    from stdnum import luhn, damm, verhoeff
+except(ImportError, IOError) as err:
+    logging.info(err)
+
+
+class TestSequenceCheckDigit(common.TransactionCase):
+
+    def test_check_digit(self):
+        sequence_obj = self.env['ir.sequence']
+        sequence = sequence_obj.create({
+            'name': 'Test sequence',
+            'implementation': 'standard',
+            'check_digit_formula': 'none',
+            'padding': '5'
+        })
+        sequence.check_digit_formula = 'luhn'
+        self.assertTrue(luhn.validate(sequence.next_by_id()))
```

Comment:
It is better to break your tests up into small methods that only test one thing. In this way, failures down the line are more apparent and you get a more clear picture of things that are wrong. When one fails and they are all in the same method, none of the other tests run, which restricts your vision IMO.